code/game/objects/items/toys.dm

```diff
@@ -178,6 +178,27 @@
 	icon = 'icons/obj/singularity.dmi'
 	icon_state = "singularity_s1"
 
+/obj/item/toy/spinningtoy/suicide_act(mob/living/carbon/human/user)
+	var/obj/item/bodypart/head/myhead = user.get_bodypart(BODY_ZONE_HEAD)
+	if(!myhead)
+		user.visible_message("<span class='suicide'>[user] tries consuming [src]... but [user.p_they()] [user.p_have()] no mouth!</span>") // and i must scream
+		return SHAME
+	user.visible_message("<span class='suicide'>[user] consumes [src]! It looks like [user.p_theyre()] trying to commit suicicide!</span>")
+	playsound(user, 'sound/items/eatfood.ogg', 50, TRUE)
+	user.adjust_nutrition(50) // mmmm delicious
+	sleep(3 SECONDS)
+
+	var/obj/item/bodypart/chest/CH = user.get_bodypart(BODY_ZONE_CHEST)
+	if(CH.cavity_item) // if he's (un)bright enough to have a round and full belly...
+		user.visible_message("<span class='danger'>[user] regurgitates [src]!</span>") // I swear i dont have a fetish
+		user.vomit(100, TRUE, distance = 0)
+		user.adjustOxyLoss(120)
+		user.dropItemToGround(src) // incase the crit state doesn't drop the singulo to the floor
```

Comment:
Honestly, I kinda want to chuck it across the room at supersonic speeds.

Reply:
do you want distance to be random between 0 and 15?

Reply:
Sounds good.  Might also want to set vomit distance to the same amount.

Reply:
on second thought, i'm not sure if this fits